libxw/src/libxw/legacy.py

```diff
@@ -0,0 +1,34 @@
+import json
+import logging
+from dataclasses import dataclass
+from typing import IO
+from typing import Optional
+from typing import Sequence
+
+
+# TODO: Configure actual logging...
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class LegacyJson:
+    spots: str
+    ans: str
+    aClues: Sequence[str]
+    dClues: Sequence[str]
+
+    @classmethod
+    def try_load(cls, buf: IO[str]) -> Optional["LegacyJson"]:
```

Comment:
In case you haven't seen it, `IO[T]` is essentially just a generic type that you can call `read()` on to get a `T` value.  This is useful for us because fileobj's (the thing you get back from `open(path)`) are `IO[str]`, and you can also turn regular `str`s into `IO[str]`.  This is just the most flexible input type for these kinds of APIs in my experience :nerd_face:

Reply:
Also, I'm returning an `Optional[T]` here instead of just a "plain" `T` because I want callers to have to think about what should happen if we fail to load from `buf` (it's likely we'll dump untrusted/unreliable input in here at some point).  Putting this information in the type system just means `mypy` can check our assumptions for us.